### ISSUE TITLE
Question on heatmap

### ISSUE THREAD

**Issue by Sirius083** [Lyken17/SparseNet#9]
Hi
Since the original densenet paper and sparseNet paper both include the heatmap part to show feature reuse, I have two corresponding questions
(1) why is the output range in (0,1) since I trained the depth_40_gr_12 on cifar10, the average absolute output of convolutional output maximum is around 0.1. I did not  get the idea of (normalized by the number of input feature maps), I use the following code to generate the correlation matrix
```python
# suppose the convolution filter shape is (3,3,128,32)
 np.mean(np.absolute(sess.run(w_tensor)), axis = (0,1,3))
```
The paper mentioned the average absolute weights are rescaled between 0 and 1 for each layer. 
The densenet paper mentioned that normalized by number of input feature maps
Does that mean the heatmap matrix is computed by
score = (x/c-min(L))/(max(L) - min(L))
where c means the input channel number
L mean the list of weights sum in a group
```python
# calculate the correlation matrix
# cifar100 depth_40_k_12_no_bottleneck
w_name = 'stage_1/dense_layer.' + str(i) + '/inner/conv/W:0'
w_tensor = graph.get_tensor_by_name(w_name)
w_value = np.mean(np.absolute(sess.run(w_tensor)), axis = (0,1,3))
c = int((w_value.shape[0]-24)/12)
w_split = [24] + [24 + 12 *(i+1) for i in range(c-1)]
res = np.array_split(w_value, w_split, axis = 0)
if len(res) == 1:
  corr[:len(res_sum),i] = res_sum
else:
  res_sum = [np.sum(d)/len(d) for d in res]
  res_sum = (res_sum - min(res_sum))/(max(res_sum) - min(res_sum))
  corr[:len(res_sum),i] = res_sum
```
Did I misunderstand the meaning?
(2) Why not take batch normalization and relu take into account?
Thanks in advance
